src/fidesops/schemas/saas/saas_config.py

```diff
@@ -104,6 +104,37 @@ def validate_request_for_pagination(cls, values: Dict[str, Any]) -> Dict[str, An
             pagination_strategy.validate_request(values)
         return values
 
+    @root_validator
+    def validate_grouped_inputs(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate that grouped_inputs must reference fields from the same collection"""
+        grouped_inputs = values.get("grouped_inputs")
+
+        if grouped_inputs:
+            request_params = values.get("request_params", [])
+            names = [param.name for param in request_params]
+
+            if not all(field in names for field in grouped_inputs):
+                raise ValueError(
+                    "Grouped_input fields must also be declared as request_params."
+                )
+
+            referenced_collections: List[str] = []
+            for param in request_params:
+                if param.name in grouped_inputs:
+                    if not param.references and not param.identity:
+                        raise ValueError(
+                            "Grouped input fields must be on incoming references or identity fields."
+                        )
+                    if param.references:
+                        collect = param.references[0].field.split(".")[0]
+                        referenced_collections.append(collect)
+            if not all(x == referenced_collections[0] for x in referenced_collections):
```

Comment:
I'm not sure if this one is just preference, something like `len(set(referenced_collections)) > 1` might be more readable

Reply:
Do you not like the `all` function? length checking seems an older way to do this, but it's not a big deal to change.

Reply:
It's not `all()` I dislike here, the comprehension took me a couple reads to grok. Then again it was late at night, so happy if you'd rather leave it in

Reply:
definitely appreciate second opinions on readability, changed it to length, thanks!